### PR TITLE
Resolve queries instead of hardcoding the path

### PR DESCRIPTION
**Pull request by koesie10**
This will change the data extensions editor generator to resolve the queries based on the `modelgenerator` tag. This removes the requirement of having a `ql` folder in the workspace.

I chose to use the path instead of the `id` for now to avoid having to resolve the query 4 times. This also avoids the need to map the language names to the language ID in the tag (i.e. `csharp` -> `cs`).

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/src/data-extensions-editor/generate-flow-model.ts [hunk 1]
```diff
@@ -16,44 +16,83 @@ import {
...
   queryRunner: QueryRunner;
   queryStorageDir: string;
-  qlDir: string;
   databaseItem: DatabaseItem;
   progress: ProgressCallback;
   token: CancellationToken;
   onResults: (results: ModeledMethodWithSignature[]) => void | Promise<void>;
 };
 
+export async function resolveQueries(
```
Comment (charisk) [inline extensions/ql-vscode/src/data-extensions-editor/generate-flow-model.ts:34]: This function doesn't need exporting.

**Review by charisk — APPROVED** [review]
The code looks good to me but would be good to get 👀 from @starcke on the approach.

File: extensions/ql-vscode/src/data-extensions-editor/generate-flow-model.ts [hunk 2]
```diff
@@ -16,45 +16,84 @@ import {
...
+      postfix: ".qls",
+    })
+  ).path;
+  const suiteYaml = [];
+  for (const qlpack of packsToSearch) {
+    suiteYaml.push({
+      from: qlpack,
+      queries: ".",
+      include: {
+        "tags contain": "modelgenerator",
```
Comment (starcke) [inline extensions/ql-vscode/src/data-extensions-editor/generate-flow-model.ts:59]: I wonder how we should ensure that the tag of these are not changed (and that not too many other queries get into this tag). Good enough for now, but lets put a note in the missing functionality doc: https://docs.google.com/document/d/1OoCEtUcShiQG328sPIoS0wNR2pqIg4Umy79UEDiSqLk/edit#heading=h.9b4gwygshmfg

**Review by starcke — APPROVED** [review]
I think this is ok, and not worse than what we were doing before. Later we should establish some agreement with the maintainers of those queries.

**Pull request merged**